From 3bb6cb939d80a99a0712f0840d65b2db912c5e74 Mon Sep 17 00:00:00 2001
From: nian0114 <268078545@qq.com>
Date: Sun, 28 May 2017 22:12:08 +0800
Subject: [PATCH] fix network type(1/2)

[*]remember to add array.xml.part to fix wrong id
---
 .../MobileNetworkSettings.smali                    | 38 +++++++++++++++++++---
 1 file changed, 33 insertions(+), 5 deletions(-)

--- a/smali/com/meizu/connectivitysettings/MobileNetworkSettings.smali
+++ b/smali/com/meizu/connectivitysettings/MobileNetworkSettings.smali
@@ -713,7 +713,7 @@

     move-result-object v2

-    const v3, 0x7f0e00b5
+    const v3, #com.meizu.connectivitysettings:array@preferred_network_mode_4g3g2g#t

     invoke-virtual {v2, v3}, Landroid/content/res/Resources;->getStringArray(I)[Ljava/lang/String;

@@ -753,14 +753,34 @@

     :sswitch_data_0
     .sparse-switch
-        0x0 -> :sswitch_1
+        0x15 -> :sswitch_1
         0x1 -> :sswitch_0
-        0x15 -> :sswitch_2
+        0x16 -> :sswitch_2
     .end sparse-switch

     :pswitch_data_0
     .packed-switch 0x0
         :pswitch_0
+        :pswitch_0
+        :pswitch_0
+	:pswitch_0
+	:pswitch_0
+	:pswitch_0
+	:pswitch_0
+	:pswitch_0
+	:pswitch_0
+	:pswitch_2
+	:pswitch_2
+	:pswitch_0
+	:pswitch_0
+	:pswitch_0
+	:pswitch_0
+	:pswitch_1
+	:pswitch_1
+	:pswitch_2
+	:pswitch_1
+	:pswitch_2
+	:pswitch_2
         :pswitch_1
         :pswitch_2
     .end packed-switch
@@ -2253,9 +2273,9 @@
     .locals 10

     .prologue
-    const v9, 0x7f0e00b7
+    const v9, #com.meizu.connectivitysettings:array@preferred_network_mode_4g3g2g_value#t

-    const v8, 0x7f0e00b5
+    const v8, #com.meizu.connectivitysettings:array@preferred_network_mode_4g3g2g#t

     const/4 v7, 0x0

@@ -5112,6 +5132,13 @@
     .restart local v11    # "modemNetworkMode":I
     goto :goto_0

+    .end local v11    # "modemNetworkMode":I
+    :pswitch_d
+    const/16 v11, 0x16
+
+    .restart local v11    # "modemNetworkMode":I
+    goto :goto_0
+
     :cond_0
     move-object/from16 v0, p0

@@ -5920,6 +5947,7 @@
         :pswitch_0
         :pswitch_0
         :pswitch_a
+        :pswitch_d
     .end packed-switch

     :pswitch_data_1
--
2.12.2
